maths/binary_exponentiation.py

```diff
@@ -4,7 +4,7 @@
 # Time Complexity : O(logn)
 
 
-def binary_exponentiation(a, n):
+def binary_exponentiation(a: int, n: float) -> int:
```

Comment:
```suggestion
def binary_exponentiation(a: int, n: int) -> int:
```

Reply:
same issue, typecheck fails with int: https://github.com/TheAlgorithms/Python/pull/9471/commits/1e931168ed0590ed7b0278bb7613fc062c23d57f
@tianyizheng02

Reply:
@tianyizheng02 however it works fine here - https://github.com/TheAlgorithms/Python/pull/9475